src/layout/rect.rs

```diff
@@ -25,6 +25,62 @@ pub struct Rect {
     pub height: u16,
 }
 
+/// Manages row divisions within a `Rect`.
+///
+/// The `Rows` struct is an iterator that allows iterating through rows of a given `Rect`.
+pub struct Rows {
+    /// The `Rect` associated with the rows.
+    pub rect: Rect,
+    /// The y coordinate of the row within the `Rect`.
+    pub current_row: u16,
+    /// The x coordinate of the row within the `Rect`.
+    pub current_column: u16,
+}
+
+impl Iterator for Rows {
+    type Item = Rect;
+
+    /// Retrieves the next row within the `Rect`.
+    ///
+    /// Returns `None` when there are no more rows to iterate through.
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.current_row >= self.rect.y + self.rect.height {
```

Comment:
You can use `Rect::bottom()` here to make this easier to maintain and avoid panics.
```suggestion
        if self.current_row >= self.rect.bottom() {
```